runtime/onert/api/include/nnfw_experimental.h

```diff
@@ -152,4 +156,194 @@ NNFW_STATUS nnfw_push_pipeline_input(nnfw_session *session, void *inputs, void *
  */
 NNFW_STATUS nnfw_pop_pipeline_output(nnfw_session *session, void *outputs);
 
+/**
+ *  Training C APIs
+ *
+ * Training APIs are designed to be used in the following order for training
+ * 1. nnfw_train_prepare
+ * 2. nnfw_train_set_input, nnfw_train_set_expected for inputs & expected outputs
+ * 3. nnfw_train
+ * 4. nnfw_train_get_loss
+ *
+ * If you want to inference after training with the same session, you can use the following order
+ * 1. nnfw_set_input
+ * 2. nnfw_set_output
+ * 3. nnfw_run
+ */
+
+//////////////////////////////////////////////
+// Essential APIs for training
+//////////////////////////////////////////////
+
+/**
+ * @brief Training information to prepare training
+ * @todo  Add more training information
+ *        (e.g. optimizer, loss function, ...)
+ */
+typedef struct nnfw_train_info
+{
+  /** Learning rate */
+  float learning_rate = 0.001f;
```

Comment:
How about separating the train preparation info and execution info?
The `nnfw_train_prepare` does not need learning_rate information. This is needed in `nnfw_train`.

After the trained graph is generated once, the user can use this graph with various learning rate and epoch. Do we need a scenario like this?

Reply:
>How about separating the train preparation info and execution info?

I didn't consider this yet. But we may need this later. 
I supposed that we will get all parameter for training before `prepare` as our first step.

>After the trained graph is generated once, the user can use this graph with various learning rate

`nnfw_train_set_traininfo` API may be used for this.